services/provider/server/server.go

```diff
@@ -2790,3 +2810,55 @@ func zeroFieldByName(obj any, fieldName string) {
 		field.SetZero()
 	}
 }
+
+func storageClassFromExisting(
+	ctx context.Context,
+	kubeClient client.Client,
+	storageClassName string,
+	consumer *ocsv1alpha1.StorageConsumer,
+	consumerConfig util.StorageConsumerResources,
+	drRbdStorageId,
+	drCephFsId string,
+) (*storagev1.StorageClass, error) {
+	storageClass := &storagev1.StorageClass{}
+	storageClass.Name = storageClassName
+	if err := kubeClient.Get(ctx, client.ObjectKeyFromObject(storageClass), storageClass); err != nil {
+		return nil, err
+	}
+	clientProfileName := ""
+	provisionerSecretName := ""
+	nodeSecretName := ""
+	operatorNamespace := consumer.Status.Client.OperatorNamespace
+	switch storageClass.Provisioner {
+	case util.RbdDriverName:
+		clientProfileName = consumerConfig.GetRbdClientProfileName()
+		provisionerSecretName = consumerConfig.GetCsiRbdProvisionerSecretName()
+		nodeSecretName = consumerConfig.GetCsiRbdNodeSecretName()
+		util.AddLabel(storageClass, ramenDRStorageIDLabelKey, drRbdStorageId)
+	case util.CephFSDriverName:
+		clientProfileName = consumerConfig.GetCephFsClientProfileName()
+		provisionerSecretName = consumerConfig.GetCsiCephFsProvisionerSecretName()
+		nodeSecretName = consumerConfig.GetCsiCephFsNodeSecretName()
+		util.AddLabel(storageClass, ramenDRStorageIDLabelKey, drCephFsId)
```

Comment:
```suggestion
```

Reply:
any reason on why this should be removed?

Reply:
removed.